### PR TITLE
Use namespaces for flags instead of interfaces

### LINKED ISSUE #304: Should all the constant-only interfaces be namespaces instead?
It's not clear to me that there will ever be `WebGPUBufferUsage` objects, say.  And if there will not, if this is just a place to put some constants, it should probably be a namespace, not an interface.

### PR DESCRIPTION
**Pull request by kainino0x**
Fixes #304. Thanks @domenic!

Unfortunately for our CI, I think we have to wait for widlparser to be updated, and for that to be rolled into bikeshed.

***

### :boom: Error: 504 Gateway Time-out :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Sep 10, 2021, 7:45 AM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [HTML Diff Service](http://services.w3.org/htmldiff) - The HTML Diff Service is used to create HTML diffs of the spec changes suggested in a pull request.

:link: [Related URL](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fpr-preview.s3.amazonaws.com%2Fgpuweb%2Fgpuweb%2Fpull%2F1709%2F9accf4e.html&doc2=https%3A%2F%2Fpr-preview.s3.amazonaws.com%2Fkainino0x%2Fgpuweb%2Fpull%2F1709.html)

```
<html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%231709.)._
</details>

### CODE REVIEW

**Review by Kangz — APPROVED** [review]

File: explainer/index.bs [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ A typical bitflag definition looks like this:
 <xmp highlight=idl>
 typedef [EnforceRange] unsigned long GPUColorWriteFlags;
 [Exposed=Window]
```
Comment (kvark) [inline explainer/index.bs:1211]: can we remove all the Exposed= attributes now? or do they still apply to namespaces?

**Review by kvark — APPROVED** [review]

File: explainer/index.bs (hunk 1 above)
Comment (domenic) [reply explainer/index.bs:1211]: They still apply (and are required)

**Comment by domenic** [comment]
Filed https://github.com/plinss/widlparser/issues/66 on widlparser

**Review by toji — APPROVED** [review]

**Comment by ExE-Boss** [comment]
Note that lines 1237 and 1238 can now be removed: <https://github.com/gpuweb/gpuweb/blob/24929350e82dd393c9f2c0be959083708a47153b/explainer/index.bs#L1237-L1238>

**Comment by kainino0x** [comment]
@ExE-Boss sharp eye, thanks!

**Comment by Kangz** [comment]
This seems like it can be landed?

**Comment by kainino0x** [comment]
Latest bikeshed should probably have the fix. Updated and attempting auto-merge

**Comment by kainino0x** [comment]
jk, it already failed. Either the change hasn't made it to bikeshed or w3c/spec-prod isn't using the latest.

**Pull request merged**